src/mango/src/mango_idx_special.erl

```diff
@@ -96,3 +98,8 @@ end_key([{_, _, '$lte', Key}]) ->
 end_key([{'$eq', Key, '$eq', Key}]) ->
     false = mango_json:special(Key),
     Key.
+
+
+maybe_filter_by_sort_fields(Idx, SortFields, _Selector) ->
+    Cols = mango_idx:columns(Idx),
```

Comment:
Seems weird to be going back out to mango_idx from inside the callback module. Shouldn't this just be columns(Idx)?